src/components/Form.test.js

```diff
@@ -305,11 +330,209 @@ describe("Form", () => {
       screen.getByText("You can't use inappropriate words.")
     ).toBeInTheDocument();
     expect(screen.getByText("Max 500 words.")).toBeInTheDocument();
+    expect(
+      screen.getByText("Please enter a street number")
+    ).toBeInTheDocument();
   });
 
   it("with testId", () => {
     const { container } = render(<SimpleForm testId="my-form" />);
 
     expect(container.firstChild).toHaveAttribute("data-testid", "my-form");
   });
+
+  describe("calling exposed functions from render child", () => {
+    it("sets form state correctly when setErrors is called", async () => {
+      const initialValues = {
+        name: "",
+        age: "",
+        address: { streetNumber: "", streetName: "" },
+      };
+
+      const initialErrors = {
+        name: ["Please enter a name"],
+        age: ["Please enter an age"],
+      };
+
+      const RenderChild = ({ setErrors, state }) => {
+        useEffect(() => {
+          if (state.values.name === "Helena") {
+            setErrors({
+              name: [
+                "This name is already taken",
+                "Try to spell it differently",
+              ],
+              "address.streetNumber": "Please enter a street number",
+            });
+          }
+        }, [setErrors, state.values.name]);
+
+        return (
+          <>
+            <Input name="name" label="Name" />
+            <Input name="age" label="Age" />
+            <Input name="address.streetNumber" label="Street number" />
+            <Input name="address.streetName" label="Street name" />
+          </>
+        );
+      };
+
+      render(
+        <Form initialValues={initialValues} initialErrors={initialErrors}>
+          {RenderChild}
+        </Form>
+      );
+
+      expect(
+        screen.queryByText("This name is already taken")
+      ).not.toBeInTheDocument();
+      expect(
+        screen.queryByText("Try to spell it differently")
+      ).not.toBeInTheDocument();
+      expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+      expect(screen.queryByText("Please enter an age")).toBeInTheDocument();
```

Comment:
This `expect` and the one above are testing the same thing that the previous test does (making sure that `initialErrors` render the errors). Let's remove these `expect`s so that each test just tests what it says it does.